### PR TITLE
Mitigating CVE-2018-1000006

### PR DESCRIPTION
**Pull request by jumde**
https://electronjs.org/blog/protocol-handler-fix
resolves  https://github.com/brave/browser-laptop/issues/12797

### CODE REVIEW

File: js/stores/appStore.js [hunk 1]
```diff
@@ -496,9 +496,10 @@ const handleAppAction = (action) => {
         if (platformUtil.isLinux()) {
           const Channel = require('../../app/channel')
           const desktopName = Channel.getLinuxDesktopName()
+          var args = [desktopName, "--"];
           for (const p of defaultProtocols) {
-            app.setAsDefaultProtocolClient(p, desktopName)
-            app.setAsDefaultProtocolClient('', desktopName)
+            app.setAsDefaultProtocolClient(p, args)
```
Comment (darkdh) [inline js/stores/appStore.js]: it should be `setAsDefaultProtocolClient(p, desktopName, ["--"])`, desktopName is not part of args

Comment (darkdh) [reply js/stores/appStore.js]: my bad, Linux implementation only reads desktop name so no need to add extra args
the args that needs -- is https://github.com/brave/browser-laptop/pull/12795/files#diff-23ca389e2bcb77191b5a9c10900eb3a3R507

Comment (darkdh) [reply js/stores/appStore.js]: https://github.com/brave/muon/blob/80442e401dd3f78d55f78daf2f5f2d8505970296/atom/browser/browser_linux.cc#L92

Comment (diracdeltas) [reply js/stores/appStore.js]: yeah this is not needed for linux afaict

File: js/stores/appStore.js [hunk 2]
```diff
@@ -496,14 +496,15 @@ const handleAppAction = (action) => {
...
+            app.setAsDefaultProtocolClient('', args)
           }
           isDefaultBrowser = app.isDefaultProtocolClient('', desktopName)
         } else {
           for (const p of defaultProtocols) {
-            app.setAsDefaultProtocolClient(p)
+            app.setAsDefaultProtocolClient(p, ["--"])
```
Comment (diracdeltas) [inline js/stores/appStore.js]: i think you need single quotes for the linter to pass

File: package.json [hunk 3]
```diff
@@ -218,7 +218,6 @@
     "plugins": [
       "flowtype"
     ],
-    "parser": "babel-eslint",
```
Comment (diracdeltas) [inline package.json]: was this not supposed to be removed?

**Comment by diracdeltas** [comment]
updated PR to do the protocol handler migration on first launch

File: js/stores/appStore.js [hunk 4]
```diff
@@ -503,7 +503,7 @@ const handleAppAction = (action) => {
           isDefaultBrowser = app.isDefaultProtocolClient('', desktopName)
         } else {
           for (const p of defaultProtocols) {
-            app.setAsDefaultProtocolClient(p)
+            app.setAsDefaultProtocolClient(p, ['--'])
```
Comment (diracdeltas) [inline js/stores/appStore.js]: doesn't the 2nd argument (`path`) need to be set? https://electronjs.org/docs/api/app#appsetasdefaultprotocolclientprotocol-path-args-macos-windows

Comment (darkdh) [reply js/stores/appStore.js]: it uses default value `Defaults to process.execPath`

Comment (darkdh) [reply js/stores/appStore.js]: we should add `['--']` just for Windows just like `protocolHandlerMigration` did

File: app/sessionStore.js [hunk 5]
```diff
@@ -868,8 +869,25 @@ const fingerprintingProtectionMigration = (immutableData) => {
...
 
+const protocolHandlerMigration = (immutableData) => {
+  // Fix https://github.com/brave/browser-laptop/issues/12797
+  if (!immutableData.get('defaultProtocolMigration')) {
+    if (platformUtil.isWindows() &&
+      immutableData.getIn(['settings', settings.IS_DEFAULT_BROWSER])) {
+      // Update the protocol handler to be safe
+      defaultProtocols.forEach((protocol) => {
+        app.removeAsDefaultProtocolClient(protocol)
+        app.setAsDefaultProtocolClient(protocol, undefined, ['--'])
```
Comment (darkdh) [inline app/sessionStore.js]: function signature here is not right `app.setAsDefaultProtocolClient(protocol, [path, args])`
https://github.com/brave/muon/blob/80442e401dd3f78d55f78daf2f5f2d8505970296/atom/browser/browser.h#L84

**Comment by darkdh** [comment]
https://github.com/brave/browser-laptop/blob/59159138e141e32c26959fc2276d5822f3ffa282/res/dev/braveDefaults.nsi#L66
has already appended `--` when we do packaging
but we still need muon changes to address the issue

**Pull request closed without merging**